src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -376,10 +376,22 @@ private static Block ConvertToMarkupAttributeBlock(Block block, bool isBoundNonS
             return blockBuilder.Build();
         }
 
-        private static Block RebuildChunkGenerators(Block block)
+        private static Block RebuildChunkGenerators(Block block, bool isBound)
         {
             var builder = new BlockBuilder(block);
 
+            // Don't want to rebuild unbound dynamic attributes. They need to run through the conditional attribute
+            // removal system at code generation time.
```

Comment:
What is this "conditional attribute removal system"? Attributes are now removed at runtime.